### PR TITLE
Workspaces block and Bindingmode block

### LINKED ISSUE #624: Add a `workspaces` block
Feature-request: would it be hard to create a block that shows workspaces?
My rationale is that I would like to have the workspaces in the same style as the rest of the bar. But currently, that would mean I'd have to get swaybar to support the same theming as i3-status which it AFAIK does not.

If there is a chance to get such a block merged, I might try to implement it myself.

### PR DESCRIPTION
**Pull request by mox-mox**
This PR should close #624. Comments/discussion welcome.

### CODE REVIEW

**Comment by mox-mox** [comment]
Please note that this PR also contains a version bumb for swayipc (2.7.2 -> 3.0.0-alpha.1).

File: blocks.md [hunk 1]
```diff
@@ -1547,6 +1572,44 @@ Key | Value
...
+
+Key | Values | Required | Default
+----|--------|----------|--------
+`strip_workspace_numbers` | Whether to remove the workspace number when displaying a workspace. | No | `false`
+`strip_workspace_name` | Whether to remove the workspace name when displaying a workspace. | No | `false`
+
+If neither `strip_workspace_numbers` nor `strip_workspace_name` are set, workspace number and name are shown, e.g. the workspace in the example would show as `1 `. Seting either options removes one part of the display, setting both options results in an empty display for each workspace. If a workspace has only the stripped part, e.g. `1` with `strip_workspace_name`, it will show that part, lest there would be nothing shown.
+
+#### TODO
+Find out what the Move, Reload and Rename events do and how to react to them.
```
Comment (ammgws) [inline blocks.md:1609]: Still WIP?

File: blocks.md [hunk 2]
```diff
@@ -190,6 +192,29 @@ Placeholder | Description
 
 ###### [↥ back to top](#list-of-available-blocks)
 
+## Bindingmode
+Creates a block which shows the binding mode in i3/sway compatible window managers. That is for example, when you press `$mod+r` to start resizing windows.
+
+When no special mode is active, i.e. the state is `default`, nothing will be shown. If another mode is triggered, the block will show a button with the name of the mode. Clicking on the button will reset the mode to `default`.
```
Comment (ammgws) [inline blocks.md:198]: By button I assume you mean button widget?

File: blocks.md (hunk 1 above)
Comment (mox-mox) [reply blocks.md:1609]: Well, there are these TODOs. Beside that, I'd think at least one person should confirm it actually works?

File: blocks.md (hunk 2 above)
Comment (mox-mox) [reply blocks.md:198]: Yes. Should I change that to read button widget?

**Comment by ammgws** [comment]
Sorry I haven't had time to look at this but thought I'd let you know that work on i3/i3#3818 seems to be progressing: https://github.com/i3/i3/pull/4311

**Comment by ammgws** [comment]
Going to close this for now.

 i3/i3#4311 looks like it's pretty much done, perhaps you could test it out and let the author know there's some interest in it?

**Pull request closed without merging**